lib/models.js

```diff
@@ -349,6 +368,14 @@ class Namespace {
   clone() {
     return new Namespace(this._namespace, this._description);
   }
+
+  toJSON() {
+    return {
+      "name": this.namespace,
+      "description": this.description,
+      "external_dependencies" : undefined
```

Comment:
That was my intention. From what I can tell, an undefined key isn't ultimately outputted unless overridden to have an actual definition, so this was more to have it documented in the tooling.

Reply:
There is a difference, in that "external_dependencies" will be a key in the object, as opposed to not defining it at all.  In other words,
```js
console.log({ "foo": "bar", "baz": undefined });
```
produces different output than:
```js
console.log({ "foo": "bar" });
```

Reply:
Whether or not that difference is meaningful will depend on the use case.  It very well might not matter here.  As long as you've confirmed it _doesn't_ matter here, then I'm fine with it.

Reply:
The console logs may be different, but the JSON output between the two is identical, so the difference isn't significant in this case.

Reply:
Additionally, I remember now, that an `undefined` field will not be translated to JSON; however, a `null` or otherwise empty field will output to `"baz": ""`, which resulted in unnecessary verbosity. Especially as key existence was intended to be a reliable method of identifying if a property exists.

Reply:
I guess what I'm saying here is that when you do `"baz": undefined"`, it results in `baz` being returns as one of the keys.

![image](https://user-images.githubusercontent.com/2278253/34744720-2f49dd82-f55c-11e7-9858-f5511308fc5c.png)

That said, it looks like it does get dropped when you call `JSON.stringify`:

![image](https://user-images.githubusercontent.com/2278253/34744812-73d4a69e-f55c-11e7-8c7e-c53c008ab113.png)

I'm really fine either way.  If you're sure it doesn't affect your code, go ahead and keep it.  We're going to stringify it when we persist to files anyway.